components/camera/videosource/webcam.go

```diff
@@ -217,48 +216,125 @@ func makeConstraints(conf *WebcamConfig, debug bool, logger logging.Logger) medi
+				" (closest driver found by gostream supports resolution %dx%d)",
+				conf.Width, conf.Height, img.Bounds().Dx(), img.Bounds().Dy())
+		}
+	}
+	return source, nil
+}
+
+// getPathFromVideoSource is a helper function for finding and making the video source that
+// returns the path derived from the underlying driver via MediaSource or an empty string if a path is not found.
+func getPathFromVideoSource(src gostream.VideoSource, logger logging.Logger) string {
+	labels, err := gostream.LabelsFromMediaSource[image.Image, prop.Video](src)
+	if err != nil || len(labels) == 0 {
+		logger.Errorw("could not get labels from media source", "error", err)
+		return ""
+	}
+
+	return labels[0] // path is always the first element
+}
+
 // findAndMakeVideoSource finds a video device and returns a video source with that video device as the source.
 func findAndMakeVideoSource(
 	ctx context.Context,
 	conf *WebcamConfig,
-	label string,
+	path string,
 	logger logging.Logger,
 ) (gostream.VideoSource, string, error) {
 	mediadevicescamera.Initialize()
 	debug := conf.Debug
 	constraints := makeConstraints(conf, debug, logger)
-	if label != "" {
-		cam, err := tryWebcamOpen(ctx, conf, label, false, constraints, logger)
+	if path != "" {
+		cam, err := tryWebcamOpen(ctx, conf, path, false, constraints, logger)
 		if err != nil {
 			return nil, "", errors.Wrap(err, "cannot open webcam")
 		}
-		return cam, label, nil
+		return cam, path, nil
 	}
 
 	source, err := gostream.GetAnyVideoSource(constraints, logger)
 	if err != nil {
 		return nil, "", errors.Wrap(err, "found no webcams")
 	}
 
-	if label == "" {
-		label = getLabelFromVideoSource(source, logger)
+	if path == "" {
+		path = getPathFromVideoSource(source, logger)
 	}
 
-	return source, label, nil
+	return source, path, nil
 }
 
-// getLabelFromVideoSource returns the path from the camera or an empty string if a path is not found.
-func getLabelFromVideoSource(src gostream.VideoSource, logger logging.Logger) string {
-	labels, err := gostream.LabelsFromMediaSource[image.Image, prop.Video](src)
-	if err != nil || len(labels) == 0 {
-		logger.Errorw("could not get labels from media source", "error", err)
-		return ""
-	}
+// webcam is a video driver wrapper camera that ensures its underlying source stays connected.
+type webcam struct {
+	resource.Named
+	mu                      sync.RWMutex
+	hasLoggedIntrinsicsInfo bool
+
+	underlyingSource gostream.VideoSource
+	exposedSwapper   gostream.HotSwappableVideoSource
+	exposedProjector camera.VideoSource
 
-	return labels[0]
+	// this is returned to us as a label in mediadevices but our config
+	// treats it as a video path.
+	targetPath string
+	conf       WebcamConfig
+
+	cancelCtx               context.Context
+	cancel                  func()
+	closed                  bool
+	disconnected            bool
+	activeBackgroundWorkers sync.WaitGroup
+	logger                  logging.Logger
+	originalLogger          logging.Logger
```

Comment:
Can we try to just have one?

Reply:
I guess the separation is used for log search when path/label changes during reconnects.

```
c.logger = c.originalLogger.WithFields("camera_label", c.targetPath)
```

I think we can preserve  the log structure for queryablity with multiple fields. something like:

```
logger: logger.WithFields(logrus.Fields{
    "camera_name":  conf.ResourceName().ShortName(),
    "camera_label": c.targetPath,
}),
```

Reply:
Talked to @dgottlieb  and we now have the adaptor support between the zap logger and our rdk logger i.e. we can assign the return value of `WithFields` to `logger` instead of having to use `originalLogger` to do so.